Tests/RSocketCoreTests/Frames/PayloadFrameTests.swift

```diff
@@ -0,0 +1,198 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import NIO
+import XCTest
+@testable import RSocketCore
+
+final class PayloadFrameTests: XCTestCase {
+
+    static private let bytes: [UInt8] = [0, 0, 0, 1,
+                                         41, 224, 0, 0,
+                                         1, 109, 100]
+
+    /* Test Payload Frame initialisation.
+     * Verify that the frame is initialised with expected header and payload.
+     */
+    func testPayloadInit() {
+        let payloadMetadata = "test payload metadata".data(using: .utf8)!
+        let payloadData = "test payload data".data(using: .utf8)!
+        let payloadFrame = PayloadFrame(streamId: 1, fragmentsFollow: true, isCompletion: true, isNext: true, payload: Payload(metadata: payloadMetadata, data: payloadData))
+
+        XCTAssertEqual(payloadFrame.header.type, .payload, "Expected payload frame type")
+        XCTAssertEqual(payloadFrame.header.streamId, 1)
+        XCTAssert(payloadFrame.header.flags.rawValue & FrameFlags.payloadFollows.rawValue != 0, "Expected follows flag to be set")
+        XCTAssert(payloadFrame.header.flags.rawValue & FrameFlags.payloadComplete.rawValue != 0, "Expected complete flag to be set")
+        XCTAssert(payloadFrame.header.flags.rawValue & FrameFlags.payloadNext.rawValue != 0, "Expected next flag to be set")
+        XCTAssert(payloadFrame.header.flags.rawValue & FrameFlags.metadata.rawValue != 0, "Expected metadata flag to be set")
+        XCTAssertEqual(payloadFrame.payload.data, payloadData)
+        XCTAssertEqual(payloadFrame.payload.metadata, payloadMetadata)
+    }
+
+    /* Test Payload Frame Encoder.
+     * Encode a payload frame and check the bytes in encoded byte buffer.
+     */
+    func testPayloadEncoder() {
+        let payloadMetadata = "m".data(using: .utf8)!
+        let payloadData = "d".data(using: .utf8)!
+        let payloadFrame = PayloadFrame(streamId: 1, fragmentsFollow: true, isCompletion: true, isNext: true, payload: Payload(metadata: payloadMetadata, data: payloadData))
+
+        guard var byteBuffer = try? PayloadFrameEncoder().encode(frame: payloadFrame, using: ByteBufferAllocator()) else {
+            XCTFail("Expected byte buffer to be not nil")
+            return
+        }
+
+        XCTAssertEqual(byteBuffer.readableBytes, 11)
+        XCTAssertEqual(byteBuffer.readBytes(length: byteBuffer.readableBytes), PayloadFrameTests.bytes)
+    }
```

Comment:
This is just a suggestion and fine if we don't apply it, but we could also define the enclosing method to be throwing:
```suggestion
    func testPayloadEncoder() throws {
        let payloadMetadata = "m".data(using: .utf8)!
        let payloadData = "d".data(using: .utf8)!
        let payloadFrame = PayloadFrame(streamId: 1, fragmentsFollow: true, isCompletion: true, isNext: true, payload: Payload(metadata: payloadMetadata, data: payloadData))

        var byteBuffer = try PayloadFrameEncoder().encode(frame: payloadFrame, using: ByteBufferAllocator())

        XCTAssertEqual(byteBuffer.readableBytes, 11)
        XCTAssertEqual(byteBuffer.readBytes(length: byteBuffer.readableBytes), PayloadFrameTests.bytes)
    }
```